src/utils/DateUtils.ts

```diff
@@ -1,24 +1,125 @@
 import { Options } from '../misc/Options';
 import { Utils } from './Utils';
 import { l } from '../strings/Strings';
-import { TimeSpan } from './TimeSpanUtils';
-import * as Globalize from 'globalize';
 import * as _ from 'underscore';
-
+import * as moment from 'moment';
+/**
+ * The 'DateUtils' class renders dates and times using momentJS using the right culture languages and formats.
+ *
+ * This class can transform strings, numbers and date objects into a standardized ISO_8601 Date.
+ *
+ */
 export interface IDateToStringOptions {
+  /**
+   * Specifies whether the date and time are the current ones.
+   */
   now?: Date;
+
+  /**
+   * Specifies whether the date should be displayed normally or using Today, Yesterday and Tomorrow.
+   *
+   * **Examples**
+   *
+   * - If the date to render is equal to the current date, it will display 'Today':
+   *    Current date: march 8 2017, date to render: march 8 2017, result: 'Today'
+   *
+   * - If the date to render is equal to the current date plus or minus 1, it will display Yesterday or Tomorrow:
+   *    Current date: march 8 2017, date to render: march 7 2017, result: 'Yesterday'
+   *
+   * Default value is 'true'
+   */
   useTodayYesterdayAndTomorrow?: boolean;
+
+  /**
+   * Specifies whether the date should be displayed normally or using days of the week if the date it is part of the
+   * current week.
+   *
+   * **Example**
+   *
+   * -If the date to render has less than a week's difference with the current date, it will display the corresponding
+   * day:
+   *    Current date: monday march 8 2017, date to render: march 11 2017, result: 'Next Thursday'
+   *
+   * Default value is 'true'
+   */
   useWeekdayIfThisWeek?: boolean;
+
+  /**
+   * Specifies whether the date should be displayed normally or without the year.
+   *
+   * **Example**
+   *
+   * - If the date to render is the same year as the current one, it won't display the year value:
+   *    Current date: march 8 2017, date to render: 09/22/2017, result: 'September 22'
+   *
+   * Default value is 'true'
+   */
   omitYearIfCurrentOne?: boolean;
+
+  /**
+   * Specifies whether the date should be displayed normally or displayed using the complete format.
+   *
+   * **Example**
+   *
+   * - If the date to render is 08/04/2017:
+   *    The result will be Friday, August 04, 2017
+   *
+   * Default value is 'false'
+   */
   useLongDateFormat?: boolean;
+
```

Comment:
Shouldn't `includeTimeIfToday`, `includeTimeIfThisWeek`, and `alwaysIncludeTime` be grouped under a single string option, instead of having three separate boolean options? It seems to me like these options override each other (always >= week >= today).

We could have an option like `includeTime?: string` and add something like "Possible values are: `onlyIfToday`, `onlyIfThisWeek`, and `always`" in the doc.

I'll wait for an answer before I suggest any doc modifications for these options.